### PR TITLE
Updated to GitHubSDK 0.4.1, now using retrofit responses.

### PR DESCRIPTION
**Pull request by Meisolsson**
Note: Just a quick port, no error handling yet implemented.

### CODE REVIEW

File: app/src/main/java/com/github/pockethub/android/core/PageIterator.java [hunk 1]
```diff
@@ -56,11 +57,11 @@ public void remove() {
 
     @Override
     public List<V> next() {
-        if(!this.hasNext()) {
+        if (!this.hasNext()) {
```
Comment (veyndan) [inline app/src/main/java/com/github/pockethub/android/core/PageIterator.java]: I'm not sure whether I'm just being nit picky, but does such a styling change reflect the intent of the PR?

Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/core/PageIterator.java]: I've done a couple of those, right now I rather fix the things when I'm close to them. No it's not exactly what the PR is intended to do but it keeps the code to the checkstyle.

Comment (veyndan) [inline app/src/main/java/com/github/pockethub/android/core/PageIterator.java]: OK I get you. It's just that doing this makes it harder to look over commits.

**Comment by HunkD** [comment]
This pr opens the possibility to check response.isSuccessful() and do error handling in the callback, right?

On the other hand, I'm curious that what's the current way to handle http error or failed response.

**Comment by veyndan** [comment]
The current way is `onError(...)` is called on response failure. Discussion on this can be found at Meisolsson/GitHubSdk#8.

**Comment by HunkD** [comment]
@veyndan thanks, the linked discussion content is very useful.

File: app/src/main/java/com/github/pockethub/android/core/PageIterator.java (hunk 1 above)
Comment (Meisolsson) [inline app/src/main/java/com/github/pockethub/android/core/PageIterator.java]: Alright, I'll try to think about it in the future.

File: app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java [hunk 2]
```diff
@@ -110,13 +117,14 @@ public void refresh() {
...
+        User result = (User) l.getItemAtPosition(position);
         ServiceGenerator.createService(getContext(), UserService.class)
                 .getUser(result.login())
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .compose(this.<User>bindToLifecycle())
-                .subscribe(user -> startActivity(UserViewActivity.createIntent(user)));
+                .compose(this.bindToLifecycle())
+                .subscribe(
+                        response -> startActivity(UserViewActivity.createIntent(response.body())));
```
Comment (veyndan) [inline app/src/main/java/com/github/pockethub/android/ui/search/SearchUserListFragment.java]: Shouldn't `response ->` be on the same line as `subscribe` to keep consistent with coding style e.g. [here](https://github.com/pockethub/PocketHub/blob/60843d315f98599ad4946106e80d0eaf52827567/app/src/main/java/com/github/pockethub/android/util/PermissionsUtils.java#L45)?

I.e.

- `.subscribe(response ->`
2.                 `startActivity(UserViewActivity.createIntent(response.body())));`

File: app/src/main/java/com/github/pockethub/android/ui/repo/RepositoryContributorsFragment.java [hunk 3]
```diff
@@ -106,8 +110,9 @@ public void onListItemClick(ListView l, View v, int position, long id) {
                 .getUser(contributor.login())
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .compose(this.<User>bindToLifecycle())
-                .subscribe(user -> startActivity(UserViewActivity.createIntent(user)));
+                .compose(this.bindToLifecycle())
+                .subscribe(
+                        response -> startActivity(UserViewActivity.createIntent(response.body())));
```
Comment (veyndan) [inline app/src/main/java/com/github/pockethub/android/ui/repo/RepositoryContributorsFragment.java]: Shouldn't response -> be on the same line as subscribe to keep consistent with coding style e.g. here?

I.e.

.subscribe(response ->
            `startActivity(UserViewActivity.createIntent(response.body())));`

**Pull request merged**